src/Apps/Order/Routes/__tests__/Shipping.test.tsx

```diff
@@ -186,12 +210,66 @@ describe("Shipping", () => {
       const component = getWrapper(testProps)
       const mockCommitMutation = commitMutation as jest.Mock<any>
       mockCommitMutation.mockImplementationOnce((_environment, config) => {
+        component.update()
         expect(config.variables.input.shipping.name).toBe("Dr Collector")
       })
-
       component.find(Button).simulate("click")
 
       expect.hasAssertions()
     })
   })
+  describe("Validations", () => {
+    let shipOrderProps
+    beforeEach(() => {
+      commitMutation.mockReset()
+      const shipOrder = {
+        ...UntouchedOrder,
+        requestedFulfillment: {
+          __typename: "Ship",
+        },
+      }
+      shipOrderProps = { ...testProps, order: shipOrder }
+    })
+
+    it("does not submit an empty form for a SHIP order", () => {
+      const component = getWrapper(shipOrderProps)
+      const { addressLine1, ...badAddress } = validAddress
+      component.find(Button).simulate("click")
+      expect(commitMutation).not.toBeCalled()
+    })
+
+    it("does not submit the mutation with an incomplete form for a SHIP order", () => {
+      const component = getWrapper(shipOrderProps)
+      const { addressLine1, ...badAddress } = validAddress
+      fillAddressForm(component, badAddress)
```

Comment:
Instead of using `fillAddressForm`, I think it may be better to just use `fillIn` so it'll be clearer what's provided and what's not and you don't have to deconstruct a `validAddress`.